pkg/model/components/apiserver.go

```diff
@@ -97,29 +97,31 @@ func (b *KubeAPIServerOptionsBuilder) BuildOptions(cluster *kops.Cluster) error
 	}
 	c.Image = image
 
-	switch cluster.GetCloudProvider() {
-	case kops.CloudProviderAWS:
-		c.CloudProvider = "aws"
-	case kops.CloudProviderGCE:
-		c.CloudProvider = "gce"
-	case kops.CloudProviderDO:
-		c.CloudProvider = "external"
-	case kops.CloudProviderHetzner:
-		c.CloudProvider = "external"
-	case kops.CloudProviderOpenstack:
-		c.CloudProvider = "openstack"
-	case kops.CloudProviderAzure:
-		c.CloudProvider = "azure"
-	case kops.CloudProviderScaleway:
-		c.CloudProvider = "external"
-	case kops.CloudProviderMetal:
-		c.CloudProvider = "external"
-	default:
-		return fmt.Errorf("unknown cloudprovider %q", cluster.GetCloudProvider())
-	}
-
-	if clusterSpec.ExternalCloudControllerManager != nil {
-		c.CloudProvider = "external"
+	if b.controlPlaneKubernetesVersion.IsLT("1.33") {
+		switch cluster.GetCloudProvider() {
+		case kops.CloudProviderAWS:
+			c.CloudProvider = "aws"
+		case kops.CloudProviderGCE:
+			c.CloudProvider = "gce"
+		case kops.CloudProviderDO:
+			c.CloudProvider = "external"
+		case kops.CloudProviderHetzner:
+			c.CloudProvider = "external"
+		case kops.CloudProviderOpenstack:
+			c.CloudProvider = "openstack"
+		case kops.CloudProviderAzure:
+			c.CloudProvider = "azure"
+		case kops.CloudProviderScaleway:
+			c.CloudProvider = "external"
+		case kops.CloudProviderMetal:
+			c.CloudProvider = "external"
+		default:
+			return fmt.Errorf("unknown cloudprovider %q", cluster.GetCloudProvider())
+		}
```

Comment:
i'm not sure how this `c.CloudProvider` value is used, but we do not allow any value other than `external` for the `--cloud-provider` flag.